src/setupWorker/start/utils/createMessageChannel.ts

```diff
@@ -11,36 +11,20 @@ export interface ServiceWorkerMessage<
   payload: EventPayload
 }
 
-export interface WorkerMessageChannel {
-  send<Event extends keyof ServiceWorkerFetchEventMap>(
-    message: Parameters<ServiceWorkerFetchEventMap[Event]>[0] extends undefined
-      ? { type: Event }
-      : {
-          type: Event
-          payload: Parameters<ServiceWorkerFetchEventMap[Event]>[0]
-        },
-  ): void
+interface WorkerChannelEventsMap {
+  MOCK_RESPONSE: (data: SerializedResponse<any>, body?: [ArrayBuffer]) => void
+  NOT_FOUND: () => void
+  NETWORK_ERROR: (data: { name: string; message: string }) => void
```

Comment:
I think we can simplify the types to represent only the body and have transfer optionally present.

like so:
```ts
interface WorkerChannelEventsMap {
  MOCK_RESPONSE: SerializedResponse<any>
  NOT_FOUND: void
  NETWORK_ERROR: { name: string; message: string }
}
```

and use it with WorkerChannel like:

```ts
  public postMessage<Event extends keyof WorkerChannelEventsMap>(
    event: Event,
    data: WorkerChannelEventsMap[Event],
    transfer?: [ArrayBuffer]
  ): void {
    this.port.postMessage({ type: event, data }, { transfer })
  }
```

Reply:
That's a great idea! We gain no benefit of keeping events map values as functions, as the return type is disregarded. Will adjust this.

Reply:
Done!